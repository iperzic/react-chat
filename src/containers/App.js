import { connect } from 'react-redux';

import * as msgActions from '../actions/messages';

import Chat from '../components/Chat';

const mapStateToProps = state => ({
  messages: state.messages,
  isLoading: state.loading,
});

const mapDispatchToProps = dispatch => ({
  onChatLoad: () => dispatch(msgActions.getMessages()),
  onNewMessage: message => dispatch(msgActions.sendMessage(message)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Chat);
