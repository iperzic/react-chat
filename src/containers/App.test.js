import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import App from './App';
import TestStore from '../support/store';

describe('App', () => {
  let store;
  beforeEach(() => {
    store = configureStore()(TestStore.createMessagesStore());
  });

  describe('when the there are no messages', () => {
    beforeEach(() => {
      store = configureStore()(TestStore.createMessagesStore());
    });
    it('should show empty input and empty message list', () => {
      const appContainer = shallow(<App store={store} />).dive();

      expect(appContainer.getElement()).toMatchSnapshot();
    });
  });

  describe('when the there are messages', () => {
    beforeEach(() => {
      store = configureStore()(TestStore.createMessagesStore(undefined, undefined, [{
        id: 'abc', author: 'Bugs Bunny', message: 'Whats up doc?', timestamp: 'Thu Mar 15 2018', isOwn: false,
      }]));
    });
    it('should show empty input and message list', () => {
      const appContainer = shallow(<App store={store} />).dive();

      expect(appContainer.getElement()).toMatchSnapshot();
    });
  });
});
