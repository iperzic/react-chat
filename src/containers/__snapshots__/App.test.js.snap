// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when the there are messages should show empty input and message list 1`] = `
<div
  className="Chat"
>
  
  <MessageList
    messages={
      Array [
        Object {
          "author": "Bugs Bunny",
          "id": "abc",
          "isOwn": false,
          "message": "Whats up doc?",
          "timestamp": "Thu Mar 15 2018",
        },
      ]
    }
  />
  <div
    className="Chat__last-message-handle"
  />
  <NewMessage
    onMessageSend={[Function]}
  />
</div>
`;

exports[`App when the there are no messages should show empty input and empty message list 1`] = `
<div
  className="Chat"
>
  
  <MessageList
    messages={Array []}
  />
  <div
    className="Chat__last-message-handle"
  />
  <NewMessage
    onMessageSend={[Function]}
  />
</div>
`;
